tests/System.Text.Primitives.Encoding.Performance.Tests/EncodingPerfComparisonTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.Utf8;
+using Microsoft.Xunit.Performance;
+
+
+namespace System.Text.Primitives.Encoding.Performance.Tests
+{
+    public class EncodingPerfComparisonTests
+    {
+        [Benchmark]
+        public void EncodingPerformanceTestUsingCoreCLR()
+        {
+            string unicodeString = GenerateString(1000);
+            ReadOnlySpan<char> characters = unicodeString.Slice();
+
+            Text.Encoding utf8 = Text.Encoding.UTF8;
+
+            int utf8Length = utf8.GetByteCount(unicodeString);
+            byte[] utf8Buffer = new byte[utf8Length];
+            Span<byte> span;
+
+            char[] charArray = characters.ToArray();
```

Comment:
> Could you also add a test where you convert span to array inside the test loop?

If you were to use the current encoding APIs, it should use the unsafe overloads that take pointers to avoid the unnecessary copy and array allocation.